src/main/java/pw/chew/chanserv/util/PluralKitLookup.java

```diff
@@ -0,0 +1,43 @@
+package pw.chew.chanserv.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.time.Instant;
+
+/**
+ * Handles basic lookup for PK's API
+ * @author Oliver-makes-code
+ * */
+public class PluralKitLookup {
+    private static final String BASE_URL = "https://api.pluralkit.me/v2/";
+
+    // Precatuion for rate limiting
+    private static Instant lastPkLookup = Instant.EPOCH;
+
+    // Checks if a message has been proxied by PK
+    public static boolean isMessageProxied(String messageId) {
+        try {
```

Comment:
Please use `RestClient` for this (it also gives us a user agent) :)

This seems to be `String data = RestClient.get(BASE_URL+"messages/"+messageId)` then `JSONObject json = new JSONObject(data)`